modin/core/io/text/text_file_dispatcher.py

```diff
@@ -552,6 +552,13 @@ def _define_metadata(
         num_splits : int
             The maximum number of splits to separate the DataFrame into.
         """
+        df = cls.read_callback(
```

Comment:
this destroys the "seek position" in the file if that is a buffer, looks weird

Reply:
There are a few problems here - it uses a variable that was originally from the user, not one that was preprocessed (`filepath_or_buffer_md`), and also that in the case of a buffer - we are trying to get the filename from which it was read (it seems the problem was that the buffer is not a serializable object), i.e. non-obvious variable name.